### DISCUSSION TITLE
Is it possible for a [Generator] to add attributes to a property / field / method / property?

### DISCUSSION THREAD

**Discussion by mrpmorris** [dotnet/roslyn#63333] [General]
I'd like to be able to add a single `[FirstName]` to a property, and have a code generator add

```c#
[MinLength(1)]
[MaxLength(32)]
[Required]
```

Is this possible?

**Comment by jaredpar** [comment]
It is not possible to _directly_ modify a member in any way and that includes adding attributes. Generally for this type of scenario our recommendation is that the user declares the member as `partial` and the generator creates a `partial` definition where the attribute is present. That won't work for fields though as they cannot be partial.

**Comment by mrpmorris** [comment]
I can't define partial properties though, can I?

**Reply by jaredpar** [reply]
Sorry yes that's correct. Those are in design but not available at the momen.

**Reply by mrpmorris** [reply]
Does this therefore mean there is no way I can get Roslyn to add additional attributes to a property?

**Reply by jasonmalinowski** [reply]
At the moment, then, no.

**Reply by mrpmorris** [reply]
I'll go with the TypeDescriptor approach then.  Thanks!

**Reply by RikkiGibson** [reply]
dotnet/csharplang#3412
